test/ipfs_mock.go

```diff
@@ -339,7 +346,14 @@ func (m *IpfsMock) handler(w http.ResponseWriter, r *http.Request) {
 			Ref: arg,
 		}
 		j, _ := json.Marshal(resp)
-		w.Write(j)
+		if arg == SlowCid1.String() {
+			for i := 0; i <= 5; i++ {
+				time.Sleep(2 * time.Second)
+				w.Write(j)
```

Comment:
Can you make sure the testing hits the missing lines here: https://codecov.io/gh/ipfs/ipfs-cluster/compare/e523215ee29b833c7cfdfb9315dd4a825a292f12...33602b9188b2edb621b9443f3c77dd016c2fccd3/diff#D1-367

The written object needs to have a `Progress` field that increases on every write.